src/CsvPayload.php

```diff
@@ -13,23 +15,102 @@ class CsvPayload extends Payload implements MiddlewareInterface
     protected $mimetype = 'text/csv';
 
     /**
-     * {@inheritdoc}
+     * The field delimiter (one character only).
+     *
+     * @var string
      */
-    protected function parse(StreamInterface $stream)
+    protected $delimiter = ",";
+
+    /**
+     * The field enclosure (one character only).
+     *
+     * @var string
+     */
+    protected $enclosure = "\"";
+
+    /**
+     * The field escape (one character only).
+     *
+     * @var string
+     */
+    protected $escape = "\\";
+
+    /**
+     * @var bool
+     */
+    private $assoc = true;
+
+    /**
+     * Configure the returned object to be converted into
+     *  - a sequential array of all CSV lines
+     *  - or a SplTempFileObject
+     *
+     * @param bool $assoc
+     *
+     * @return self
+     */
+    public function associative($assoc = true)
```

Comment:
For styling purposes: the method name should edit a variable with the same name. Example:

```php
public function associative($associative = true)
{
    //...
}
public function delimiter($delimiter)
{
    //...
}